src/parallel/MPI_TIMED/mpi_timed.cpp

```diff
@@ -0,0 +1,226 @@
+	ProcessTimer::get().startTimer();
+	int result = PMPI_Test(request, flag, status);
+	ProcessTimer::get().stopTimer();
+	return result;
+}
+
+int MPI_Type_create_struct(int count, const int array_of_blocklengths[], const MPI_Aint array_of_displacements[],
+                           const MPI_Datatype array_of_types[], MPI_Datatype * newtype) {
+	ProcessTimer::get().startTimer();
+	int result = PMPI_Type_create_struct(count, array_of_blocklengths, array_of_displacements, array_of_types, newtype);
+	ProcessTimer::get().stopTimer();
+	return result;
+}
+
+int MPI_Type_commit(MPI_Datatype * datatype) {
+	ProcessTimer::get().startTimer();
+	int result = PMPI_Type_commit(datatype);
+	ProcessTimer::get().stopTimer();
+	return result;
+}
+
+int MPI_Type_free(MPI_Datatype *datatype) {
+	ProcessTimer::get().startTimer();
+	int result = PMPI_Type_free(datatype);
+	ProcessTimer::get().stopTimer();
+	return result;
+}
+
+int MPI_Cart_create(MPI_Comm comm_old, int ndims, const int dims[], const int periods[], int reorder,
+                    MPI_Comm * comm_cart){
+	ProcessTimer::get().startTimer();
+	int result = PMPI_Cart_create(comm_old, ndims, dims, periods, reorder, comm_cart);
+	ProcessTimer::get().stopTimer();
+	return result;
+}
+
+int MPI_Cart_coords(MPI_Comm comm, int rank, int maxdims, int coords[]) {
+	ProcessTimer::get().startTimer();
+	int result = PMPI_Cart_coords(comm, rank, maxdims, coords);
+	ProcessTimer::get().stopTimer();
+	return result;
+}
+
+int MPI_Cart_rank(MPI_Comm comm, const int coords[], int *rank) {
+	ProcessTimer::get().startTimer();
+	int result = PMPI_Cart_rank(comm, coords, rank);
+	ProcessTimer::get().stopTimer();
+	return result;
+}
+
+int MPI_Op_free(MPI_Op * op) {
+	ProcessTimer::get().startTimer();
+	int result = PMPI_Op_free(op);
+	ProcessTimer::get().stopTimer();
+	return result;
+}
+
+int MPI_Op_create(MPI_User_function *user_fn, int commute, MPI_Op *op) {
+	ProcessTimer::get().startTimer();
+	int result = PMPI_Op_create(user_fn, commute, op);
+	ProcessTimer::get().stopTimer();
+	return result;
+}
+
+int MPI_Get_count(const MPI_Status * status, MPI_Datatype datatype, int *count) {
+	ProcessTimer::get().startTimer();
+	int result = PMPI_Get_count(status, datatype, count);
+	ProcessTimer::get().stopTimer();
+	return result;
+}
+
+int MPI_Comm_free(MPI_Comm * comm) {
+	ProcessTimer::get().startTimer();
+	int result = PMPI_Comm_free(comm);
+	ProcessTimer::get().stopTimer();
+	return result;
+}
+
+int MPI_Get_address(const void *location, MPI_Aint * address) {
+	ProcessTimer::get().startTimer();
+	int result = PMPI_Get_address(location, address);
+	ProcessTimer::get().stopTimer();
+	return result;
+}
+
+int MPI_Dims_create(int nnodes, int ndims, int dims[]) {
+	ProcessTimer::get().startTimer();
+	int result = PMPI_Dims_create(nnodes, ndims, dims);
+	ProcessTimer::get().stopTimer();
+	return result;
+}
+
+int MPI_Finalize(void) {
+	int result = PMPI_Finalize();
+	// ProcessTimer::get().writeProcessTimeLog();
```

Comment:
why is this commented out? Can it be removed or should this be enabled in some debug mode?